Extensions/TweenBehavior/tweenruntimebehavior.ts

```diff
@@ -58,119 +100,102 @@ namespace gdjs {
+        tweenable,
         false,
         startTime,
         totalDuration
       );
-    }
 
-    _getTween(identifier) {
-      return this._tweens[identifier];
+      // Attach it to the scene as it will become active and acttive tweens should be on the scene.
+      if (!this._runtimeScene.shiftyJsScene)
+        this._runtimeScene.shiftyJsScene = new shifty.Scene();
+      this._runtimeScene.shiftyJsScene.add(tweenable);
+
+      // Start the tween and set the needed callbacks
+      tweenable
+        .tween(tweenConfig)
+        .then(() => {
+          if (this._tweens[identifier])
+            this._tweens[identifier].hasFinished = true;
+
+          if (destroyObjectWhenFinished)
+            this.owner.deleteFromScene(this._runtimeScene);
+        })
+        .catch((e) => {
+          logger.error('An error has occured while running the tween: ', e);
+        });
     }
 
-    _tweenExists(identifier) {
+    private _tweenExists(identifier: string): boolean {
       return !!this._tweens[identifier];
     }
 
-    _tweenIsPlaying(identifier) {
-      return this._tweens[identifier].instance.isPlaying();
+    private _tweenIsPlaying(identifier: string): boolean {
+      return (
+        this._tweens[identifier] &&
+        this._tweens[identifier].instance.isPlaying()
+      );
     }
 
-    _pauseTween(identifier) {
+    private _pauseTween(identifier: string): void {
       const tween = this._tweens[identifier];
 
       // Pause the tween, and remove it from the scene of living tweens
       // (the invariant is that scene only contains tweens being played).
       tween.instance.pause();
 
-      // @ts-ignore - shiftyJsScene is added to runtime scene.
       const shiftyJsScene = this._runtimeScene.shiftyJsScene;
       if (shiftyJsScene) {
         shiftyJsScene.remove(tween.instance);
       }
     }
 
-    _resumeTween(identifier) {
+    private _resumeTween(identifier: string): void {
       const tween = this._tweens[identifier];
 
       // Resume the tween, and add it back to the scene of living tweens
       // (the invariant is that scene only contains tweens being played).
       tween.instance.resume();
 
-      // @ts-ignore - shiftyJsScene is added to runtime scene.
       const shiftyJsScene = this._runtimeScene.shiftyJsScene;
       if (shiftyJsScene) {
         shiftyJsScene.add(tween.instance);
       }
     }
 
-    _stopTween(identifier, jumpToDest) {
-      return this._tweens[identifier].instance.stop(jumpToDest);
-    }
-
-    _setTweenFinished(identifier, hasFinished) {
-      this._tweens[identifier].hasFinished = hasFinished;
+    private _stopTween(identifier: string, jumpToDest: boolean): void {
+      this._tweens[identifier].instance.stop(jumpToDest);
     }
 
-    _tweenHasFinished(identifier) {
+    private _tweenHasFinished(identifier: string): boolean {
       return this._tweens[identifier].hasFinished;
     }
 
-    _removeObjectFromScene(identifier) {
-      this._removeTween(identifier);
```

Comment:
Weird that we were calling _removeTween - not sure what was the reason or if this was useful 🤔

Reply:
I'm not sure myself after all this time. ^^
I think the tween wasn't deleted automatically when it finished.

Reply:
Well if the object is removed the tween not being removed shouldn't matter, right?

Reply:
I agree this shoud not matter - just asking to see if there was not a hidden issue. I think it's safe to remove this line!